includes/multi-currency/BackendCurrencies.php

```diff
@@ -0,0 +1,171 @@
+<?php
+/**
+ * WooCommerce Payments Multi Currency Backend Currencies
+ *
+ * @package WooCommerce\Payments
+ */
+
+namespace WCPay\MultiCurrency;
+
+use WC_Payments_Localization_Service;
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * Class that formats Multi Currency currencies on the backend.
+ */
+class BackendCurrencies {
+	/**
+	 * Multi-Currency instance.
+	 *
+	 * @var MultiCurrency
+	 */
+	protected $multi_currency;
+
+	/**
+	 * WC_Payments_Localization_Service instance.
+	 *
+	 * @var WC_Payments_Localization_Service
+	 */
+	protected $localization_service;
+
+	/**
+	 * Multi-Currency currency formatting map.
+	 *
+	 * @var array
+	 */
+	protected $currency_format = [];
+
+	/**
+	 * Constructor.
+	 *
+	 * @param MultiCurrency                    $multi_currency       The MultiCurrency instance.
+	 * @param WC_Payments_Localization_Service $localization_service The Localization Service instance.
+	 */
+	public function __construct( MultiCurrency $multi_currency, WC_Payments_Localization_Service $localization_service ) {
+		$this->multi_currency       = $multi_currency;
+		$this->localization_service = $localization_service;
+
+		// We need to check first if it's a request coming from the backend, frontend REST requests shouldn't be
+		// affected by this.
+		$is_backend_request = 0 === strpos( strtolower( wp_get_referer() ), strtolower( admin_url() ) );
+
+		// Add the filter if it's an admin request or a REST request from the admin side.
+		if ( ( is_admin() || ( WC()->is_rest_api_request() && $is_backend_request ) ) && ! defined( 'DOING_CRON' ) ) {
+			// Currency hooks. Be aware that this should not run after Explicit Price hook, its priority should be less
+			// than explicit price hooks to run before them.
+			add_filter( 'wc_price_args', [ $this, 'build_wc_price_args' ], 50 );
+
+		}
+	}
+
+	/**
+	 * Returns the store currency code.
+	 *
+	 * @return string The store currency code
+	 */
+	public function get_store_currency(): string {
+		return get_option( 'woocommerce_currency' );
```

Comment:
Should we be using `get_woocommerce_currency()` here so that if for some reason the currency is filtered we are in sync with everything else?

Reply:
Sure, I'll change that. IIRC, I copied it from somewhere else :) I'll fix it.